checks.d/splunk_topology.py

```diff
@@ -0,0 +1,190 @@
+"""
+    StackState.
+    Splunk topology extraction
+"""
+
+# 3rd party
+import requests
+from urllib import quote
+import time
+
+# project
+from checks import AgentCheck, CheckException
+
+
+class SavedSearch:
+    def __init__(self, saved_search):
+        self.name = saved_search['name']
+        self.element_type = saved_search['element_type']
+        self.parameters = saved_search['parameters']
+
+
+class InstanceConfig:
+    def __init__(self, instance, init_config):
+        default_timeout = init_config.get('default_timeout', 5)
+        max_retry_count = init_config.get('max_retry_count', 3)
+        seconds_between_retries = init_config.get('seconds_between_retries', 1)
+        polling_interval = init_config.get('polling_interval', 15)
+
+        self.base_url = instance['url']
+        self.username = instance['username']
+        self.password = instance['password']
+        self.timeout = float(instance.get('timeout', default_timeout))
```

Comment:
I've added some comments in the config file and changed the name a bit.
Regarding the 'global' timeout, I would like that, however the script is single threaded; you block for a http request before being able to check for this global timeout possibly incurring more wait time before actually timing out (overhead of max request timeout). Some explaining is then needed for the two different timeouts if we go that route. Let's discuss furher tomorrow.